sdcm/nemesis.py

```diff
@@ -168,16 +169,16 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
     limited: bool = False           # flag that signal that nemesis are belong to limited set of nemesises
     has_steady_run: bool = False    # flag that signal that nemesis should be run with perf tests with steady run
 
```

Comment:
Whole this change looks as overkill for solving problem you described.
If you need to avoid duplicates then, I think, would be easier to add attribute that will track names of disruptive method that were applied and run wrapper only if no record in such a mapping.

Example:

```
class Nemesis:
    ...
    wrapped_disruptive_methods = []
    ...

    def __new__(cls, tester_obj, termination_event, *args):  # pylint: disable=unused-argument
        for name, member in inspect.getmembers(cls, lambda x: inspect.isfunction(x) or inspect.ismethod(x)):
            if name.startswith(cls.DISRUPT_NAME_PREF) and name not in cls.wrapped_disruptive_methods:
                # add "disrupt_method_wrapper" decorator to all methods are started with "disrupt_"
                setattr(cls, name, disrupt_method_wrapper(member))
                wrapped_disruptive_methods.append(name)
        return object.__new__(cls)
```

So, in such a case we have 2 new lines and 1 updated.

Reply:
This is other solution for the problem.
May prevent multiply wrapping.
changed